ID-B-4.3

roman-net8 opened the pull request:
Отрефакторил сущности Volunteer и Pet добавил ValueObjects (объекты значения) record Requisite - уже реализовано
record SocialNetwork - уже реализовано
record Address - добавил

установил библиотеку CSharpFunctionalExtensions, и внес изменения в проект для Result.

roman-net8 committed:
Б-4.3
Отрефакторил сущности Volunteer и Pet добавил ValueObjects (объекты значения)
record Requisite - уже реализовано
record SocialNetwork - уже реализовано
record Address - добавил

установил библиотеку CSharpFunctionalExtensions, и внес изменения в проект для Result.

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/Address.cs:
```diff
@@ -0,0 +1,19 @@
+
+public record Address
+{
+    private Address(string country, string city, string street, string houseNumber, string appartmentNumber)
```
каждый параметр на своей строке

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/Address.cs:
```diff
@@ -0,0 +1,19 @@
+        HouseNumber = houseNumber;
+        AppartmentNumber = appartmentNumber;
+    }
+
```
Сделать валидацию и фабричный метод create

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/FullName.cs:
```diff
@@ -2,6 +2,16 @@
+    {
+        FirstName = firstName;
+        MiddleName = middleName;
+        LastName = lastName;
```
Сделать валидацию и фабричный метод create

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/Requisite.cs:
```diff
@@ -2,6 +2,16 @@
+    private Requisite(string title, string description)
+    {
+        Title = title;
+        Description = description;
```
Сделать валидацию и фабричный метод create

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/FullName.cs:
```diff
@@ -2,6 +2,16 @@
 
 public record FullName
 {
+    private FullName()
```
У VO только 1 конструктор, убрать

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/Requisite.cs:
```diff
@@ -2,6 +2,16 @@
 
 public record Requisite
 {
+    private Requisite()
```
У VO только 1 конструктор, убрать

roman-net8 committed:
исправленя по комментариям в PetFamily/pull/9
добавил "PetDetails" исправил Requisite и Pet

YADARAN1 commented on backend/src/PetFamily.Domain/Models/Pets/PetDetails.cs:
```diff
@@ -0,0 +1,10 @@
+{
+    private readonly List<Requisite> _petRequisites = [];
+
+    public IReadOnlyList<Requisite> PetRequisites => _petRequisites;
```
В обёртку достаточно написать public IReadOnlyList<Requisite> PetRequisites { get; }

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/FullName.cs:
```diff
@@ -1,18 +1,43 @@
+
+    public string FirstName { get; } = default!;
+    public string MiddleName { get; } = default!;
+    public string LastName { get; } = default!;
```
Добавь nullable, потому-что у некоторых людей возможно нет отчества

YADARAN1 commented on backend/src/PetFamily.Domain/Models/Pets/PetRequisites.cs:
```diff
@@ -0,0 +1,14 @@
+        Requisites = requisites.ToList();
+    }
+
+    public IReadOnlyList<Requisite> Requisites { get; } = [];
```
Так лучше не объявлять = [], в будущем могут быть проблемы

roman-net8 committed:
исправления по комментариям в PetFamily/pull/9
добавил -  nullable для FullName.MiddleName
исправил -  public IReadOnlyList PetRequisites { get; }
